File: back/src/main/java/com/pingo/dto/UserSignUp.java
package com.pingo.dto;

import com.pingo.entity.users.UserInfo;
import com.pingo.entity.users.Users;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class UserSignUp {

    private Users users;
    private UserInfo userInfo;
    private String userMyKeyword;
    private String userFavoriteKeyword;
}

File: front/lib/data/models/sign_model/user_signup.dart
import 'package:pingo_front/data/models/user_model/user_info.dart';
import 'package:pingo_front/data/models/user_model/users.dart';

class UserSignup {
  Users users;

  UserInfo userInfo;

  String? userMyKeyword;
  String? userFavoriteKeyword;

  UserSignup({
    Users? users,
    UserInfo? userInfo,
    this.userMyKeyword,
    this.userFavoriteKeyword,
  })  : users = users ?? Users(),
        userInfo = userInfo ?? UserInfo();

  factory UserSignup.createEmpty() {
    return UserSignup(
      users: Users(),
      userInfo: UserInfo(),
      userMyKeyword: '',
      userFavoriteKeyword: '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "users": users.toJson(),
      "userInfo": userInfo.toJson(),
      "userMyKeyword": userMyKeyword,
      "userFavoriteKeyword": userFavoriteKeyword,
    };
  }

  @override
  String toString() {
    return 'UserSignup{users: $users, userInfo: $userInfo, userMyKeyword: $userMyKeyword, userFavoriteKeyword: $userFavoriteKeyword}';
  }
}
